system/jlib/jevent.hpp

```diff
@@ -115,6 +116,43 @@ struct jlib_decl EventRecordingSummary
     StringBuffer filename;
 };
 
+interface IEventAttribute
+{
+    virtual EventAttr queryId() const = 0;
+    virtual bool isText() const = 0;
+    virtual bool isNumeric() const = 0;
+    virtual bool isBoolean() const = 0;
+    virtual const char* queryTextValue() const = 0;
+    virtual __uint64 queryNumericValue() const = 0;
+    virtual bool queryBooleanValue() const = 0;
+};
+
+interface IEventAttributeIterator : extends IInterface
+{
+    virtual bool first() = 0;
+    virtual bool next() = 0;
+    virtual bool isValid() const = 0;
+    virtual const IEventAttribute& query() const = 0;
+};
+
+interface IEvent : extends IInterface
```

Comment:
I would avoid an interface, and have a concrete class to avoid unnecessary overhead creating and reading these objects.